models/model.js

```diff
@@ -24,3 +24,21 @@ exports.fetchArticlesFromDB = () => {
     return results.rows;
   });
 };
+
+
+exports.fetchArticlebyIdFromDB = (id) => {
+    if (!/\d/.test(id)) {
+      return Promise.reject({ status: 400, message: "Bad request" });
```

Comment:
Great custom err handling, alternatively PSQL will throw an error back if the id is not a number, you could handle it with a PSQL handler in your middleware file. I think the err code is 22P02. 
Handling this elsewhere will clean up your code and you won't need regex!

Reply:
It's near the bottom: 

https://notes.northcoders.com/courses/js-back-end/error-handling